### PR TITLE
vdoc: refactor theme files

### PR DESCRIPTION
**Pull request by ttytm**
I'd like to submit further improvements to sites that users can generate with `v doc`. I hope it's okay to refactor the theme files to an es6 standard beforehand. Without breaking the general style of the code. Just to update the syntax and to allow for a cleaner base when working with the files.

This PR:

* Uses `let`, `const` variable declarations, ternary expressions, arrow functions for anonymous function arguments.
* Updates loop syntax, uses methods where appropriate
* Reduces unnecessary querying / querying selectors from parent elements if they are already known
* Removes some obsolete code only relevant for Internet Explorer. IE usage is below 0.2%. Also asking Microsofts Bing AI:
  > Today website support of the Internet Explorer browser is not required, even Microsoft stopped its support. It is better to take care of the correct support and display of your website on other browsers, for example, Google Chrome. ...
* Adds a formatter config for the `vdoc/theme/` directory to ensure formatting consistency as styles are currently a bit mixed up. I used prettier for this as it's the most common for web formatting and it covers all the theme files. 

Since this is a batch of changes, I've tried to structure them into pickable commits.

### <samp>🤖 Generated by Copilot at 444aca0</samp>

This pull request improves the JavaScript code quality and functionality of the `vdoc` tool's theme files. It refactors `doc.js` to use modern and consistent syntax and features, and fixes some dark mode and scroll spy issues. It also simplifies the code to get the html element in `dark-mode.js` by using `querySelector`.

### <samp>🤖 Generated by Copilot at 444aca0</samp>

*  Simplify the code to get html elements by using `querySelector` instead of `getElementsByTagName` or `getElementsByClassName` ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-c2bf169d4f36325e12ebc945a62818daad2d9982e7db28fd8298a7c51d6d2f2eL2-R3), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL17-R36), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL67-R59), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL74-R70), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL89-R86))
*  Refactor the code to use `const` and `let` declarations instead of `var` declarations, to use more modern and consistent syntax for immutable and mutable variables ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL3-R9), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL17-R36), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL54-R53), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL67-R59), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL74-R70), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL89-R86), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL111-R110), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL123-R119), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL135-R144), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL159-R157), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL181-R184), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL208-R195), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL219-R202), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL231-R214))
*  Refactor the code to use arrow functions instead of regular functions, to make the code more concise and consistent ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL54-R53), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL74-R70), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL89-R86))
*  Move the `setupDarkMode` function call before the `setupScrollSpy` function call, to ensure that the dark mode is applied before the scroll spy logic is executed ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL3-R9))
*  Use the optional chaining operator `?.` to avoid errors if the `active` element is null or undefined ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL3-R9))
*  Use the `Array.from` and `map` methods to create the `sectionPositions` array from the `sections` collection, instead of using a `forEach` loop ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL17-R36))
*  Remove a `console.log` statement that seems to be used for debugging purposes only ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL54-R53))
*  Add a `docNav` variable to store the reference to the `.doc-nav` element, to avoid repeating the query selector multiple times ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL89-R86))
*  Simplify the sorting function by using a ternary operator instead of an if-else statement ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL135-R144))
*  Use `for-of` loops instead of regular `for` loops, to iterate over arrays without using indexes ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL159-R157), [link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL208-R195))
*  Change the name of the `ul` variable to `ul_`, to avoid confusion with the previous `ul` variable in the same scope ([link](https://github.com/vlang/v/pull/19024/files?diff=unified&w=0#diff-3fb25060e841bc44786d7d6ed02a91a4c6c44f16784bb57b6af7e7934f57eb0eL159-R157))

### CODE REVIEW

**Comment by JalonSolov** [comment]
Make sure that the changes you make work on Firefox as well.  There is still a lot of support for it, and it isn't built on Chrome, so there could be differences.

**Comment by ttytm** [comment]
> Make sure that the changes you make work on Firefox as well. There is still a lot of support for it, and it isn't built on Chrome, so there could be differences.

I'll do my best. Went through supporting IE6 when doing more webdev a while ago. Still have nightmares about it.

**Comment by JalonSolov** [comment]
IE6 was definitely a nightmare.  Firefox isn't... it's very standards compliant.  Chrome seems to have some specific extensions that make pages not work quite right with Firefox, but I haven't seen the reverse - it if works with Firefox, it seems to work the same with Chrome.

File: cmd/tools/vdoc/theme/dark-mode.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
-(function() {
```
Comment (danieldaeschle) [inline cmd/tools/vdoc/theme/dark-mode.js:1]: You can also use arrow functions if you upgrade to the newest browsers.

File: cmd/tools/vdoc/theme/doc.js [hunk 2]
```diff
@@ -1,235 +1,204 @@
...
-        var content = document.querySelector('.doc-nav .content');
-        content.classList.toggle('hidden');
-        content.classList.toggle('show');
-    });
+	document.getElementById('toggle-menu').addEventListener('click', () => {
+		const docNav = document.querySelector('.doc-nav');
+		const isHidden = docNav.classList.contains('hidden');
+		docNav.classList.toggle('hidden');
+		const search = docNav.querySelector('.search');
+		// console.log(search);
```
Comment (danieldaeschle) [inline cmd/tools/vdoc/theme/doc.js:52]: Remove this console.log

**Review by danieldaeschle — APPROVED** [review]

**Review by spytheman — APPROVED** [review]
Excellent work.

File: cmd/tools/vdoc/theme/dark-mode.js (hunk 1 above)
Comment (ttytm) [reply cmd/tools/vdoc/theme/dark-mode.js:1]: The two iffes in the code were the parts I was most unsure about. Afaik it works without the function wrapper so I assumed they just encapsulated the initialization code in a block  - for a readability reason?

I had this changed for a moment to an arrorw function, but as they are on the same level as  the function declarations I left the `function` keyword. To extend my elaboration a little bit: for normal declarations I came to prefer `function myFn() {` to `const myFn = () => {`  - after using the latter exclusively for a while. It's more immediately visible while reading that it's a function not a variable I think. Also the main way functions are declared i.e. in the Svelte framework.

File: cmd/tools/vdoc/theme/doc.js (hunk 2 above)
Comment (ttytm) [reply cmd/tools/vdoc/theme/doc.js:52]: :+1: I'll do it when adding following changes. It was there so I just commented it out here.

Thanks for the review ! :heart:

**Pull request merged**